eppo_core/src/events/event_dispatcher.rs

```diff
@@ -0,0 +1,186 @@
+use std::collections::VecDeque;
+use crate::events::event::{Event};
+use log::info;
+use tokio::sync::mpsc::{UnboundedReceiver, UnboundedSender};
+use tokio::time::{Duration, Instant};
+
+#[derive(Debug)]
+pub enum EventDispatcherCommand {
+    Event(Event),
+    Flush,
+    Exit,
+}
+
+// batch size of zero means each event will be delivered individually, thus effectively disabling batching.
+const MIN_BATCH_SIZE: usize = 0;
```

Comment:
🐛 batch sizes 0/1 are not handled and require at least one another event before sending a batch

Reply:
good call. i'll se the min batch size to 1 and make sure we deliver immediately in that case